test/boulder-tools/Dockerfile

```diff
@@ -1,17 +1,23 @@
 FROM buildpack-deps:focal-scm
-ARG GO_VERSION
 
 # Provided automatically by docker buildx.
 ARG TARGETPLATFORM
 ARG BUILDPLATFORM
 
+ENV TARGETPLATFORM=$TARGETPLATFORM
 ENV PATH /usr/local/go/bin:/usr/local/protoc/bin:$PATH
 
-RUN arch=$(echo $TARGETPLATFORM | sed 's|\/|-|') && wget -O go.tgz "https://dl.google.com/go/go${GO_VERSION}.${arch}.tar.gz" && tar -C /usr/local -xzf go.tgz && rm go.tgz;
 COPY requirements.txt /tmp/requirements.txt
 COPY boulder.rsyslog.conf /etc/rsyslog.d/
 COPY build.sh /tmp/build.sh
 RUN /tmp/build.sh
+# Note: This arg and env variable should only be set _after_ build.sh installs
+# necessary apt packages. Otherwise, we will reinstall those apt packages for
+# each Go version, rather than reusing a cached layer.
+ARG GO_VERSION
+ENV GO_VERSION=$GO_VERSION
+COPY install-go.sh /tmp/install-go.sh
+RUN /tmp/install-go.sh
 RUN sed -i '/imklog/s/^/#/' /etc/rsyslog.conf
 RUN sed -i '/$ActionFileDefaultTemplate/s/^/#/' /etc/rsyslog.conf
 RUN sed -i '/$RepeatedMsgReduction on/s/^/#/' /etc/rsyslog.conf
```

Comment:
Since build.sh is the thing that installs rsyslog, I'd move these RUNs to be immediately after build.sh rather than at the very end. I don't believe this would have negative effects on the caching behavior, and I believe it should have (negligibly) good effects on it.